device/globals/esc_compile.gd

```diff
@@ -13,9 +13,11 @@ var commands = {
 	"inventory_remove": { "min_args": 1 },
 	"inventory_open": { "min_args": 1, "types": [TYPE_BOOL] },
 	"set_active": { "min_args": 2, "types": [TYPE_STRING, TYPE_BOOL] },
+	"use_action_menu": { "min_args": 2, "types": [TYPE_STRING, TYPE_BOOL] },
 	"stop": true,
 	"repeat": true,
 	"wait": true,
+	"set_speed": { "min_args": 2, "types": [TYPE_STRING, TYPE_INT] },
```

Comment:
Might be better with a more general purpose `set_property` which takes a global_id, the property to set and the new value?

Reply:
That could replace `use_action_menu` as well.

Reply:
You'd probably have to use [Object.get()](http://docs.godotengine.org/en/3.0/classes/class_object.html#id1) to check the type of the property, cast it to the right type and call [Object.set()](http://docs.godotengine.org/en/3.0/classes/class_object.html#id4) to set the new value.

Reply:
Naturally I did consider this, but complicated checking and a lack of need made me go the other way.

I tried coming up with use cases and most of them, like changing `use_custom_z` just don't make sense, so having a general `set_property` opens up a lot of options a game developer may not want to bother with. IMO the ESC scripts should be simple.

If you want, `set_property` could be there, but I'd advocate _strongly_ for having short-hands for the most common cases _as well_ to keep it simple.

If you don't feel very strongly about this, I advocate not implementing `set_property`.